### ISSUE TITLE
NOD | Refine Sentry error monitoring dashboards

### ISSUE THREAD

**Issue by saderagsdale** [department-of-veterans-affairs/va.gov-team#58173]
## Value Statement
**_As a_** software developer
**_I want to_** refine the Sentry dashboard for the Request for Board Appeal (Notice of Disagreement)  form
**_So that_** I can easily easy spot errors that need attention and remove noise from the dashboard.

---
## Background Context
We have a Sentry dashboard for our benefits and decision reviews portfolio here: http://sentry.vfs.va.gov/organizations/vsp/dashboard/12/?environment=production. It is still a working draft. The queries for each widget need to be refined so that they capture all errors stemming from their respective forms, and exclude errors that aren't actually errors or that don't actually require follow up. Ultimately, we want this Sentry dashboard to show all true errors stemming from each form and feature in our benefits and decision reviews portfolio. This will help us understand the health of our portfolio and help us know how to allocate engineering resources toward maintenance.

N.B. This ticket is **not** about fixing the errors surfaced by the widgets for NOD. This ticket is about refining the widgets for NOD so that they show all true errors stemming from NOD. Once this ticket is complete, we will know what is wrong with NOD and can follow up accordingly.

## Acceptance Criteria
- [ ] Dashboard has refined widgets.
- [ ] All relevant stakeholders are made aware of these widgets. 

## Out of scope
- Bug fixes from errors identified as the widgets are refined. Once this ticket is complete, we will know what is wrong and can follow up accordingly.
- Notifying the platform team of systemic errors

## Tasks
- [ ] Go through front end and back end code and ensure that all code paths are covered by the widget's query. 
- [ ] Go through the list of errors produced by this query (looking at the past 90 days), and exclude errors that aren't errors or that don't actually require follow up.
- [ ] Make sure all widgets are using this same query.

---
## Definition of Ready 
- [ ] Know what to include and exclude from the dashboard
- [ ] Have an agreement on what visual representation works best for developers

## Definition of Done
- [ ] The final result is reviewed and approved by FE and BE developers
- [ ] The dashboard has been created and is successfully displaying relevant data in an easy-to-read and actionable format
- [ ] Make revisions to the [Logging and Monitoring Framework](https://docs.google.com/document/d/1TqBQmbYGF7Xvjzs_2q83qDuoKd9022n153wVl8V4MmI/edit#heading=h.wb43vjfgcjqq) as needed and report back to Jared/Kevin.
- [ ] The user story has been reviewed and approved by all relevant stakeholders, including product and development.

**Comment by saderagsdale** [comment]
@data-doge @kylesoskin Please review for backlog refinement.

**Comment by Mottie** [comment]
NOD Sentry dashboards have been updated
- [Decision reviews](http://sentry.vfs.va.gov/organizations/vsp/dashboard/10/?environment=production&statsPeriod=90d)
- [Benefits + Decision Reviews Portfolio](http://sentry.vfs.va.gov/organizations/vsp/dashboard/12/?environment=production&statsPeriod=90d)

And [added NOD sentry documentation](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/benefits-decision-reviews/monitoring/sentry-filters.md)

**Comment by data-doge** [comment]
Looks great! Thank you @Mottie. I'm going to close this ticket. Though, given your discovery that Sentry's filter queries have a 400 char max, I plan to create a follow up spike ticket to investigate transitioning this sentry monitoring dashboard to DataDog. If we're to propose a scalable error monitoring framework to VA, I think we'll want a platform with a bit more flexibility. Ticket coming soon.
